main-server.js

```diff
@@ -28,7 +30,9 @@ const defaultOptions = {
 
 export const createApolloServer = (givenOptions, givenConfig) => {
 
+  let graphiqlOptions = Object.assign({}, defaultConfig.graphiqlOptions, givenConfig.graphiqlOptions);
   let config = Object.assign({}, defaultConfig, givenConfig);
```

Comment:
If we do it that way any ```graphiqlOptions``` that we pass will override the whole ```defaultConfig.graphiqlOptions``` object. That's because ```Object.assign``` is not deep.